core/src/test/java/rdx/works/core/IdentifiedArrayListImplTest.kt

```diff
@@ -0,0 +1,70 @@
+package rdx.works.core
+
+import org.junit.Before
+import org.junit.Test
+
+
+class IdentifiedArrayListImplTest {
+
+    internal class IdentifiedString(val value: String) : Identified {
+
+        override val identifier: String
+            get() = value
+
+    }
+
+    private val identifiedArrayList = IdentifiedArrayList<IdentifiedString>()
+
+    @Before
+    fun setUp() {
+        identifiedArrayList.add(IdentifiedString("1"))
+        identifiedArrayList.add(IdentifiedString("2"))
+        identifiedArrayList.add(IdentifiedString("3"))
+        identifiedArrayList.addAll(listOf(IdentifiedString("4"), IdentifiedString("5"), IdentifiedString("5")))
+    }
+
+    @Test
+    fun `test adding duplicate element`() {
+        assert(identifiedArrayList.add(IdentifiedString("1")).not())
```

Comment:
Personally I prefer `assertFalse(expression)` since it makes it easier to read the test.